### ISSUE TITLE
Usage Warning Needed

### ISSUE THREAD

**Issue by blu-dev** [Coolsonickirby/SSBU-Dumped-Status-Scripts#2]
This repository is potentially a good resource for people who are curious to see general functionality for how a move functions, however there are some serious faults in here and this should not be advertised as a utility for newcomers to actually *reimplement* status scripts.

This is functionally equivalent to dumping the entirety of the main executable as a bunch of `.c` files and telling people to use it as a reference for creating skyline hooks/reimplementing core functionality, it's just not a good idea.

Here are some cautions I've seen so far/that have been brought to my attention (either via DMs from people asking me how to edit the status scripts or from others lamenting this repository's existence):
- Status scripts that make calls to local functions abstract away a lot of the logic that would otherwise be necessary, and unlike ACMD scripts these functions **are not** able to be assumed. An example is Lucario's `SpecialNEnd` [status](https://github.com/Coolsonickirby/SSBU-Dumped-Status-Scripts/blob/master/ghidra_scripts/lua2cpp_lucario.nro/lucario/lucario/status/SpecialN/end.c)
- Some L2CValue operator signatures are incorrect because the script used to "fix" the analyzed NROs in ghidra did not account for them at the time of authorship
- Some `L2CAgentBase` math methods have incorrect signatures
- Anything that returns on the SIMD registers is basically an alien language
- Ghidra's labels are **extremely** hard to follow and dumping them to a C file makes them even harder to relabel or figure out xrefs to.

Please place a disclaimer on this repository that this is to be used as a reference for understanding status script logic and is not in any capacity currently viable to copy/paste scripts into smashline code and have them be functional.

Thanks!

p.s. I imagine that this is to lower the barrier of entry for rewriting status scripts for fighter addition/your clone engine, please let me know if this is the case and I can help you with a better solution than expecting your users to copy/paste all of the status scripts into their own codebase.

**Comment by blu-dev** [comment]
None of this mentions the glaring issue that smashline is a Rust library, and this is pseudo-C. I applaud you for having the patience to dump all of these manually but you've effectively provided the resource that Ghidra provides without the value of any of the disassembler's tooling.

Wuboy's repos are in a format that can be directly applied, this is not and again needs a very clear and preferably large disclaimer.

EDIT: I now see that there is a link to a Ghidra guide in the README that basically says "go do it yourself if you need this other information". At that point why even have the repo?

**Comment by Coolsonickirby** [comment]
Hello!

I appreciate the time you took for providing this feedback for this repo. I apologize if this came off as a repo for people to be able to easily take and throw it in their rust code (and I can see why now as to the similar naming scheme to this repository name and WuBoy's repository), as I made this repo more so for the purpose of those who want to be able to reference/read fighters & articles without having to download Ghidra on the machine they're currently using (such as if they're borrowing a computer or are on a phone/tablet.)

I will update the readme to include the fact that this should only be used as a reference.

Regarding the edit, it's meant for users who want to dig deeper (for the case of missing functions) and gives instructions on how to connect to the shared Ghidra server which has the lua2cpp files uploaded onto it. Mostly for the case that if someone happens to stumble onto this repo and wants to explore more but doesn't know how.
